src/compiler/checker.ts

```diff
@@ -5503,6 +5503,10 @@ namespace ts {
                 return symbol.declarations && find(symbol.declarations, s => !!getEffectiveTypeAnnotationNode(s) && (!enclosingDeclaration || !!findAncestor(s, n => n === enclosingDeclaration)));
             }
 
+            function getExistingNodeHasNoTypeParametersOrMatchingTypeParameters(existing: TypeNode, type: Type) {

```

Comment:
Should this start with the word `get`?

Reply:
By convention, yes - pretty much all of our functions start with verbs, `get`, `create`, `serialize`, and so on.

Reply:
What about predicate functions like `typeHasCallOrConstructSignatures`? Doesn't seem appropriate.

Reply:
but emacs doesn't show completions until I've typed 3 characters!

Reply:
I renamed it to `existingTypeNodeIsNotReferenceOrIsReferenceWithCompatibleTypeArgumentCount` which _surely_ isn't a mouthful. I hope the removal of the verb is to your liking?

Reply:
named perfectly!